include/aws/http/request_response.h

```diff
@@ -296,6 +296,60 @@ struct aws_http_request_handler_options {
     aws_http_on_stream_complete_fn *on_complete;
 };
 
+struct aws_http1_stream_chunk;
+
+/**
+ * Signature definition of the function which will be invoked, if provided by the caller, after a chunked transfer
+ *  encoding chunk has been written to an HTTP/1.1 connection.
+ */
+typedef void aws_http1_stream_write_chunk_complete_fn(struct aws_http1_stream_chunk *chunk, void *user_data);
+
+/**
+ * HTTP/1.1 chunk extension for chunked encoding.
+ * Note that the underlying strings are not owned by the byte cursors.
+ */
+struct aws_http1_chunk_extension {
+    struct aws_byte_cursor key;
+    struct aws_byte_cursor value;
+};
+
+/**
+ * Encoding options for an HTTP/1.1 chunked transfer encoding chunk.
+ */
+struct aws_http1_chunk_options {
+    /**
+     * A pointer to an array of chunked extensions. The num_extensions must match the length of the array.
+     */
+    struct aws_http1_chunk_extension *extensions;
+
+    /**
+     * The number of elements defined in the extensions array.
+     */
+    size_t num_extensions;
+
+    /**
+     * Holds the number of extensions already sent over the connection.
+     */
+    size_t sent_extensions;
+
+    /**
+     * A caller provided callback which will be invoked after the chunk has been submitted to the channel for writing
+     * to the underlying HTTP connection.
+     */
+    aws_http1_stream_write_chunk_complete_fn *on_complete;
+
+    /**
+     * User provided data passed to the on_complete callback on its invocation.
+     */
+    void *user_data;
+};
+
+struct aws_http1_stream_chunk {
+    struct aws_input_stream *stream;
+    struct aws_http1_chunk_options options;
+    struct aws_linked_list_node node;
```

Comment:
If you end up keeping `aws_http_1_stream_chunk` part of the public API, consider prefixing some variables with "internal_" or "private_" or "_detail" or grouping them like
```
    struct {
        size_t sent_extensions;
        struct aws_linked_list node;
    } private;
};
```

Reply:
I have removed it from the public API.